### PR TITLE
Run commands and scripts to windows hosts

### PR DESCRIPTION
**Pull request by ltamaster**
Add option to run commands, scripts (Node Executor and File Copier) to windows hosts, it required winrm support on the ansible server (see http://docs.ansible.com/ansible/intro_windows.html)

Also, fix an issue getting the nodes form Ansible when the host has defined the attribute ansible_host equal to null

### CODE REVIEW

File: src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java [hunk 1]
```diff
@@ -290,12 +294,12 @@ public INodeSet getNodes() throws ResourceModelSourceException {
           // ansible_system     = Linux   = osFamily in Rundeck
           // ansible_os_family  = Debian  = osName in Rundeck
 
-          if (root.has("ansible_system")) {
-            node.setOsFamily(root.get("ansible_system").getAsString());
+          if (root.has("ansible_os_family")) {
```
Comment (frozenice) [inline src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java:297]: Are you sure this mapping for OsFamily and OsName to Ansible facts is correct (haven't tested it)?

**Comment by frozenice** [comment]
Needs a rebase / merge so the CI can run.

File: src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java (hunk 1 above)
Comment (DerekTBrown) [inline src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java]: I doubt it very much

**Review by DerekTBrown — CHANGES_REQUESTED** [review]
This would break the "choose default shell" option.  Need to add a setting for Windows users.

File: src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java (hunk 1 above)
Comment (ltamaster) [reply src/main/java/com/batix/rundeck/plugins/AnsibleResourceModelSource.java:297]: Hi, this worked for me on ansible 2.3.1.0.

screenshot 2017-08-09 22 39 52

screenshot 2017-08-10 00 27 45
screenshot 2017-08-10 00 28 36

**Comment by ltamaster** [comment]
Hi @DerekTBrown 
About the "choose default shell" option.
By default I didn't set the executor for windows hosts. If I add it (and if I add new options on "choose default shell" for windows, like cmd.exe or powershell.exe) the plugin will lose the option (by default)  to manage multiples OS on the same project (because a default value like bash/sh will fail on windows host ). Of course in this case the workaround will be that the user overwrites the ansible-executor at node level, but that would need that the user create another resource model manually.

So, I was thinking to  define a default executor on the Model Resource for windows host like:

screenshot 2017-08-10 10 57 05

Of course, the disadvantage will be if the user wants to use another default shell.

Another workaround is to add a second "choose default shell" but for windows hosts.

Please let me know what you think about it.

Thanks
Luis

**Comment by DerekTBrown** [comment]
@ltamaster I like the idea of a second option better than fixing it to Powershell, it is cleaner.

**Comment by ltamaster** [comment]
Hi @DerekTBrown, 

I added the second option for windows

screenshot 2017-08-15 16 02 45

**Comment by ltamaster** [comment]
Hi @DerekTBrown,

This is a friendly follow up, is the new settings (Windows Executable)  looks OK for you?

Thanks
Luis

**Comment by DerekTBrown** [comment]
@ltamaster looks good to me!  I know that @frozenice was working on some CI tests for this package so that we can validate further commits, so I was initially going to hold off until we could integrate this into the test suite.  However, it doesn't seem like that is going to happen soon, so I am going to merge this under the assumption that we will need to write some CI tests for Windows systems later.

**Review by DerekTBrown — APPROVED** [review]

**Pull request merged**